### ISSUE TITLE
to_pandas() | Combine filters with OR boolean operation

### ISSUE THREAD

**Issue by yohplala** [dask/fastparquet#573]
Hi,

I would like to request a feature already supported by pyarrow when reading partitions and/or row-groups: the combination of filters through OR bolean operation, not only AND.
pyarrow supports this by accepting list of lists of tuples.
 - tuples are unit filters
 - list of tuples combines each filter by AND boolean operation
 - outer list combines each list of tuples by OR boolean operation

pyarrow's `read_table()` function [doc](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html#pyarrow-parquet-read-table), see `filters` parameter.

Basically, given a time series, I would like to load all data between TS1 & TS2, and all data between TS3 & TS4, and so on...
For this, this feature is the answer I see.

Thanks for your help.
Have a good day,
Bests,

**Comment by martindurant** [comment]
There is no reason fastparquet couldn't use the same filtering logic. Note, though, that we do not filter *within* row groups - that would be substantially more work (but still doable!). I am unlikely to have any time to work on this.

**Comment by yohplala** [comment]
Hi Martin, how are you?

> Note, though, that we do not filter within row groups 

Yes, I noticed ;) (usually, I start writing a ticket once I have done some basic survey about the topic of interest ;))

> I am unlikely to have any time to work on this.

Ok, at first I was not 'boiling hot' to spend some time on the topic either, but still I read through the code and saw that it was not a big deal. Please, find a proposal in PR #574 

Please, do not hesitate to share your comments, they are welcome!
Have a good week-end,
Bests
